### PR TITLE
[8.x] ESQL: Properly skip datasets with lookup indices

### PR DESCRIPTION
**Pull request by alex-spies**
Do not rely on specific naming, read the actual settings.

This is required to unblock the backport https://github.com/elastic/elasticsearch/pull/118661, which adds a new lookup index dataset whose name doesn't end with `_lookup`.

When this is merged, I plan to manually forward port it to main; the PR is against 8.x because this is where the problems occur.

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvTestsDataLoader.java [hunk 1]
```diff
@@ -258,11 +257,22 @@ public static void main(String[] args) throws IOException {
...
+                && (supportsIndexModeLookup || isLookupDataset(dataset) == false)) {
+                testDataSets.add(dataset);
+            }
+        }
+
+        return testDataSets;
+    }
+
+    public static boolean isLookupDataset(TestsDataset dataset) throws IOException {
+        Settings settings = dataset.readSettingsFile();
```
Comment (craigtaverner) [inline x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvTestsDataLoader.java:273]: OK, that was easier than expected!

**Review by craigtaverner — APPROVED** [review]
LGTM. Not sure why the stream was removed, but I guess it looks more classic-java now!

**Comment by alex-spies** [comment]
CI seems to be fussy for unrelated reasons, like timeouts in [OldRepositoryAccessIT](https://gradle-enterprise.elastic.co/s/fx4jamxf7fjzu/tests/task/:x-pack:qa:repository-old-versions:javaRestTestAfterRestart%236_8_20/details/org.elasticsearch.oldrepos.OldRepositoryAccessIT). This change doesn't affect the other test suite at all.

**Comment by alex-spies** [comment]
Thanks for reviews @idegtiarenko and @craigtaverner !

> Not sure why the stream was removed

This makes it easier to propagate IOExceptions, streams don't like checked exceptions :/

**Pull request merged**